src/rgw/rgw_sal.h

```diff
@@ -1430,7 +1431,7 @@ class Writer : public ObjectProcessor {
                        const char *if_match, const char *if_nomatch,
                        const std::string *user_data,
                        rgw_zone_set *zones_trace, bool *canceled,
-                       optional_yield y) = 0;
+                       const req_context& rctx) = 0;
```

Comment:
these sal changes also need to be applied to the overrides in driver/daos, driver/motr and driver/posix

Reply:
ok. how do i control their compilation?
this would make sure i don't miss anything

Reply:
WITH_RADOSGW_POSIX is enabled by default. you could enable WITH_RADOSGW_MOTR and WITH_RADOSGW_DAOS, but you'd have to provide their dependencies manually. i don't think it's really worth trying to compile those, especially if your interface changes are trivial like this

Reply:
MOTR and DAOS break regularly.  I think I'm the only one who builds them.